downloader/src/main/kotlin/Main.kt

```diff
@@ -237,9 +244,54 @@ object Main {
         } else {
             p("No VCS URL provided.")
             // TODO: This should also be tried if the VCS checkout does not work.
+
             p("Trying to download source package ...")
-            // TODO: Implement downloading of source package.
-            throw DownloadException("No source code package URL provided.")
+            return downloadSourcePackage(target, outputDirectory)
+        }
+    }
+
+    private fun downloadSourcePackage(target: Package, outputDirectory: File): File {
+        if (target.sourceArtifact.url.isBlank()) {
+            throw DownloadException("No source artifact URL provided.")
+        }
+
+        val request = Request.Builder().url(target.sourceArtifact.url).build()
+
+        val response = OkHttpClientHelper.execute(HTTP_CACHE_PATH, request)
+        if (!response.isSuccessful || response.body() == null) {
+            throw DownloadException("Failed to download source artifact: $response")
+        }
+
+        val tempFile = createTempFile(suffix = target.sourceArtifact.url.substringAfterLast("/"))
+
+        tempFile.outputStream().use { stream ->
+            stream.write(response.body()!!.bytes())
+        }
+
+        verifyChecksum(tempFile, target.sourceArtifact.hash, target.sourceArtifact.hashAlgorithm)
+
+        try {
+            tempFile.unpack(outputDirectory)
+        } catch (e: IOException) {
+            log.error { "Could not unpack source artifact '${tempFile.absolutePath}': ${e.message}" }
+            throw DownloadException(e)
+        }
+
+        return outputDirectory
+    }
+
+    private fun verifyChecksum(file: File, hash: String, hashAlgorithm: String) {
+        val digest = when (hashAlgorithm.toLowerCase()) {
+            "md5" -> file.inputStream().use { DigestUtils.md5Hex(it) }
+            "sha-1" -> file.inputStream().use { DigestUtils.sha1Hex(it) }
```

Comment:
Note: We probably need to also support "sha1" (without the dash) here in the future. Maybe add that already now, like `"sha1", "sha-1" ->`.

Reply:
Didn't know that syntax, cool.